cinema/views.py

```diff
@@ -32,7 +46,27 @@ class CinemaHallViewSet(viewsets.ModelViewSet):
 
 class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.all()
-    serializer_class = MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        genres = self.request.query_params.get("genres")
+        actors = self.request.query_params.get("actors")
+        title = self.request.query_params.get("title")
+
+        if genres:
+            genres = genres.split(",")
+            queryset = queryset.filter(genres__id__in=genres)
```

Comment:
this works because Django tries to cover us, but the params type should be int here (or list of ints in this case)

Reply:
idk about it, i think if it works, thats fine.
please explain me more why i need to do changes in that example

Reply:
In Django, when you are filtering querysets based on related fields, it’s crucial to ensure that the types of values you use in the filter match the types expected by the database schema. In the example you provided:

if genres:
    genres = genres.split(",")
    queryset = queryset.filter(genres__id__in=genres)
Explanation of the Issue:
Type Mismatch:

Database Schema: In your database schema, the genres__id field is likely an integer (or a list of integers if it’s a Many-to-Many relationship).
Current Code: genres is being split from a comma-separated string, so initially, genres is a list of strings (e.g., ['1', '2', '3']).
Filtering by IDs:

When you use genres__id__in=genres, Django’s ORM expects genres to be a list of integers, as it’s filtering on an integer field (id).
Since genres is currently a list of strings, this can lead to a type mismatch or unexpected behavior.
Why You Should Change It:
Consistency with Database Schema: The id field in your database is an integer. Therefore, for accurate filtering and to avoid potential errors, you should convert the list of string IDs to a list of integers before filtering.
Avoid Runtime Errors: If Django or the database engine attempts to perform the filtering with a list of strings where integers are expected, it might result in errors or inefficient queries.